ddtrace/mocktracer/mocktracer.go

```diff
@@ -79,9 +91,31 @@ func (t *mocktracer) FinishedSpans() []Span {
 func (t *mocktracer) Reset() {
 	t.Lock()
 	defer t.Unlock()
+	t.unfinishedSpans = nil
 	t.finishedSpans = nil
 }
 
+func (t *mocktracer) addUnfinishedSpan(s Span) {
```

Comment:
I'd say to remove these two methods and instead just have a field on `*mocktracer`:

```go
openSpans map[uint64]Span
```
The key would be the `SpanID`. This will make it easy to work with and you can get rid of both `addUnfinishedSpan` and `removeUnfinishedSpan` which just makes things more complicated. Simply add and remove from the map and then in `UnfinishedSpans` (or `OpenSpans`) iterate over the map and make the slice.

Reply:
yup, that's simpler